src/moonstone-samples/lib/DataGridListSample.js

```diff
@@ -110,12 +147,72 @@ module.exports = kind({
 		this.modelIndex = idx;
 		return records;
 	},
+	generateDataGridList: function(itemType) {
+		if (this.$.gridList) {
+			this.$.gridList.destroy();
+		}
+
+		var moreProps = {};
+
+		switch (itemType) {
+		case 'HorizontalGridListImageItem':
+			moreProps = {minWidth: 600, minHeight: 100,
+				components: [{
+					kind: HorizontalGridListImageItem,
+					bindings: [
+						{from: 'model.text', to: 'caption'},
+						{from: 'model.subText', to: 'subCaption'},
+						{from: 'model.url', to: 'source'},
+						{from: 'model.selected', to: 'selected', oneWay: false}
+					]
+				}]
+			};
+			break;
+		case 'HorizontalGridListItem':
+			moreProps = {minWidth: 600, minHeight: 100,
+				components: [{
+					kind: HorizontalGridListItem,
+					bindings: [
+						{from: 'model.text', to: 'caption'},
+						{from: 'model.subText', to: 'subCaption'},
+						{from: 'model.selected', to: 'selected', oneWay: false}
+					]
+				}]
+			};
+			break;
+		default:
+			break;
+		}
+
+		var props = utils.mixin({}, [this.dataListDefaults, moreProps]);
+		var c = this.$.listPanel.createComponent(props, {owner: this});
+		c.render();
+		this.set('collection', new Collection(this.generateRecords(40)));
+
+		this.$.gridList.set('collection', this.collection);
+		this.$.gridList.set('selection', this.$.selectionToggle.value);
+		if (this.$.selectionTypeGroup.active) {
+			this.$.gridList.set('selectionType', this.$.selectionTypeGroup.active.value);
+		}
+	},
+	selectionChanged: function(inSender, inEvent) {
+		this.$.gridList.set('selection', inSender.value);
+	},
+	itemTypeChanged: function(inSender, inEvent) {
+		this.generateDataGridList(inSender.active.value);
+	},
+	selectionTypeChanged: function(inSender, inEvent) {
```

Comment:
Be sure to leave a space after `function` for all function definitions and to not prefix arguments with `in`

`selectionTypeChanged: function (sender, event) {`